controllers/etcd.go

```diff
@@ -215,11 +215,26 @@ func (r *TalosControlPlaneReconciler) auditEtcd(ctx context.Context, tcp *contro
 
 		present := false
 		for _, machine := range machines.Items {
-			// break apart the noderef name in case it's an fqdn (like in AWS)
-			machineNodeNameExploded := strings.Split(machine.Status.NodeRef.Name, ".")
+			var hostname string
 
-			if machineNodeNameExploded[0] == member.Hostname {
+			for _, address := range machine.Status.Addresses {
+				if address.Type == clusterv1.MachineHostName {
+					hostname = address.Address
+
+					break
+				}
+			}
+
+			if hostname == "" {
+				// break apart the noderef name in case it's an fqdn (like in AWS)
+				parts := strings.Split(machine.Status.NodeRef.Name, ".")
```

Comment:
I _think_ we should split both hostname/domainname... 

we can do `hostname, _ = strings.Cut(hostname, ".")`

Reply:
done